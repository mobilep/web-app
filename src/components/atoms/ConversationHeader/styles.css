.ConversationHeader {
	display: flex;
	justify-content: space-between;
	width: 100%;
}

.ConversationHeader-user {
	color: var(--grey3);
	font-size: 14px;
	display: flex;
	align-items: center;
	flex-grow: 1;
}

.ConversationHeader-user span {
	font-size: 20px;
	font-weight: 600;
	color: var(--almostBlack);
}

.ConversationHeader-user p {
	margin-bottom: 0;
	font-size: 12px;
}

.ConversationHeader-avatar {
	width: 40px;
	height: 40px;
}

.ConversationHeader-avatar + .ConversationHeader-user-name {
	margin-left: 20px;
}

.ConversationHeader-evaluate {
	flex-grow: 10;
}

.ConversationHeader-evaluate > button {
	margin: 0 5px;
}

.ConversationHeader-user-right {
	display: flex;
	align-items: center;
}

@media (max-width: 840px) {
	.ConversationHeader-user .Avatar {
		display: none;
	}
}

@media (max-width: 1140px) {
	.ConversationHeader-user-name-secondary {
		display: none;
	}
}
