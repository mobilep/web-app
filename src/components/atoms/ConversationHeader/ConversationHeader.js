import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import { Avatar, Button } from '../..';
import { AVATAR_SIZES } from '../Avatar/constants';
import { getInitials } from '../../../helpers';

import './styles.css';

class ConversationHeader extends Component {

	render () {
		const {
			title,
			actionButton,
			actionButtonLabel,
			className,
			onActionButtonClick,
			inboxMode,
			onDeleteButtonClick,
			deleteButtonLabel,
			scenarioCoach,
			scenarioUser,
			scenarioOwner,
			coach,
			user,
		} = this.props;
		const userInitials = user && getInitials(user.firstName, user.lastName);
		const coachInitials = coach && getInitials(coach.firstName, coach.lastName);

		return (
			<div className={classNames('ConversationHeader', className)}>
				<div className='ConversationHeader-user'>
					{!inboxMode &&
					<Avatar
						className='ConversationHeader-avatar'
						initials={scenarioOwner ? userInitials : coachInitials}
						size={AVATAR_SIZES.AVATAR_44}
						color={scenarioOwner ? user && user.avatarColor : coach && coach.avatarColor}
						img={scenarioOwner ? user && user.avatar_sm : coach && coach.avatar_sm}
					/>}
					<div className='ConversationHeader-user-name'>
						<span>{title}</span>
						{!inboxMode && <p>{scenarioOwner ? scenarioUser : scenarioCoach}</p>}
					</div>
				</div>
				{actionButton &&
				<div className='ConversationHeader-evaluate'>
					<Button className='Button-text p-h-0' onClick={onActionButtonClick}>
						{actionButtonLabel}
					</Button>
				</div>
				}
				{inboxMode &&
					<Button className='Button-text p-h-0' onClick={onDeleteButtonClick}>
						{deleteButtonLabel}
					</Button>
				}
				{!inboxMode && <div className='ConversationHeader-user ConversationHeader-user-right'>
					<Avatar
						className='ConversationHeader-avatar'
						initials={!scenarioOwner ? userInitials : coachInitials}
						size={AVATAR_SIZES.AVATAR_44}
						color={!scenarioOwner ? user && user.avatarColor : coach && coach.avatarColor}
						img={!scenarioOwner ? user && user.avatar_sm : coach && coach.avatar_sm}
					/>
					<div className='ConversationHeader-user-name ConversationHeader-user-name-secondary'>
						<span>{scenarioOwner ? coach && coach.name : user && user.name}</span>
						{!inboxMode && <p>{!scenarioOwner ? scenarioUser : scenarioCoach}</p>}
					</div>
				</div>}
			</div>
		);
	}

}

ConversationHeader.defaultProps = {
	title: '',
	actionButtonLabel: 'Evaluate',
	onActionButtonClick: () => {},
	onDeleteButtonClick: () => {},
};

ConversationHeader.propTypes = {
	actionButton: PropTypes.bool,
	actionButtonLabel: PropTypes.string,
	className: PropTypes.string,
	coach: PropTypes.object,
	deleteButtonLabel: PropTypes.string,
	inboxMode: PropTypes.bool,
	onActionButtonClick: PropTypes.func,
	onDeleteButtonClick: PropTypes.func,
	scenarioCoach: PropTypes.string,
	scenarioOwner: PropTypes.bool,
	scenarioUser: PropTypes.string,
	title: PropTypes.string,
	user: PropTypes.object,
};

export default ConversationHeader;
