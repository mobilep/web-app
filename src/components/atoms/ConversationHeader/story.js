import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs/react';
import ConversationHeader from './ConversationHeader';

storiesOf('1. Atoms / ConversationHeader', module)
	.add('Conversation Header', () => (
		<ConversationHeader
			title={text('title', 'Alvin Lane')}
			actionButtonLabel={text('button label', 'Evaluate')}
			onActionButtonClick={action('onActionButtonClick')}
		/>
	));
