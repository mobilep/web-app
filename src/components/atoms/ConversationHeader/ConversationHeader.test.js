import React from 'react';
import { shallow } from 'enzyme';

import ConversationHeader from './ConversationHeader';

describe('Atom - ConversationHeader', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<ConversationHeader
				title='Test'
				scenarioCoach='Jon Doe'
				scenarioUser='Test Test'
				user={{ firstName: 'Jon', lastName: 'Doe' }}
				coach={{ firstName: 'Jon', lastName: 'Doe' }}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render with action button', () => {
		const wrapper = shallow(
			<ConversationHeader
				title='Test'
				actionButton
				actionButtonLabel='Label'
				scenarioCoach='Jon Doe'
				scenarioUser='Test Test'
				user={{ firstName: 'Jon', lastName: 'Doe' }}
				coach={{ firstName: 'Jon', lastName: 'Doe' }}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
