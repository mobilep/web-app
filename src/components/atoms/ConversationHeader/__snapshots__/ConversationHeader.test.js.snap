// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - ConversationHeader Should render 1`] = `
<div
  className="ConversationHeader"
>
  <div
    className="ConversationHeader-user"
  >
    <Avatar
      border={false}
      className="ConversationHeader-avatar"
      initials="JD"
      size="avatar-44"
    />
    <div
      className="ConversationHeader-user-name"
    >
      <span>
        Test
      </span>
      <p>
        Jon Doe
      </p>
    </div>
  </div>
  <div
    className="ConversationHeader-user ConversationHeader-user-right"
  >
    <Avatar
      border={false}
      className="ConversationHeader-avatar"
      initials="JD"
      size="avatar-44"
    />
    <div
      className="ConversationHeader-user-name ConversationHeader-user-name-secondary"
    >
      <span />
      <p>
        Test Test
      </p>
    </div>
  </div>
</div>
`;

exports[`Atom - ConversationHeader Should render with action button 1`] = `
<div
  className="ConversationHeader"
>
  <div
    className="ConversationHeader-user"
  >
    <Avatar
      border={false}
      className="ConversationHeader-avatar"
      initials="JD"
      size="avatar-44"
    />
    <div
      className="ConversationHeader-user-name"
    >
      <span>
        Test
      </span>
      <p>
        Jon Doe
      </p>
    </div>
  </div>
  <div
    className="ConversationHeader-evaluate"
  >
    <Button
      className="Button-text p-h-0"
      onClick={[Function]}
    >
      Label
    </Button>
  </div>
  <div
    className="ConversationHeader-user ConversationHeader-user-right"
  >
    <Avatar
      border={false}
      className="ConversationHeader-avatar"
      initials="JD"
      size="avatar-44"
    />
    <div
      className="ConversationHeader-user-name ConversationHeader-user-name-secondary"
    >
      <span />
      <p>
        Test Test
      </p>
    </div>
  </div>
</div>
`;
